### PR TITLE
Workshop for deploying Node.js apps on heroku

### PR DESCRIPTION
**Pull request by sajedazoabi**
create a short technical workshop/guide
Deploying on heroku - a 20 minute thing (walkthrough md file)
#631

### CODE REVIEW

**Comment by eliascodes** [comment]
what issue does this address?

**Comment by sajedazoabi** [comment]
@eliascodes  - issue #631

File: coursebook/week-4/Heroku Workshop [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+```
+Now check that you have git installed. If not, install it and test again.
+```
+$ git --version
+>
+git version 2.2.1
+```
+****
+### 2. Prepare the app
+clone a repo for Node.js app from github .
```
Comment (dangerdak) [inline coursebook/week-4/Heroku Workshop]: Nice instructions. If this is a workshop, should we include an url to a repo they can clone?

File: coursebook/week-4/Heroku Workshop [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+Once dependencies are installed, you will be ready to run your app locally.
+****
+### 5. Run the app locally
+Now start your application locally using the heroku local command, which was installed as part of the Heroku CLI:
+```
+$ heroku local web
+```
+To stop the app from running locally, in the CLI, press Ctrl+C to exit.
+****
+### 6.Push local changes
```
Comment (eliascodes) [inline coursebook/week-4/Heroku Workshop]: Tell me if I'm misreading, but the instructions in this section sound like they're assuming committing to master. I don't think that's a practise we want to encourage.

File: coursebook/week-4/Heroku Workshop (hunk 1 above)
Comment (eliascodes) [reply coursebook/week-4/Heroku Workshop]: There's also a stray space before the full stop.

**Comment by sajedazoabi** [comment]
@eliascodes - First Thanks for reviewing and the useful comments.
Taking a step forward - after I asked other students and users, they think that following the basic Heroku guide contains unnecessary steps and can disturb people who don't have English as Native language.
So This workshop is very short and direct.

File: coursebook/week-4/Heroku Workshop.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
+# Deploying Node.js Apps on Heroku
+***This Workshop assumes that you have a free Heroku account, and that you have Node.js and npm installed locally.***
+****
+## What is Heroku?
+Heroku is a cloud platform as a service (PaaS) supporting several programming languages that is used as a web application deployment model. Heroku supports Ruby, Java, Node.js, Scala, Clojure, Python and PHP programming languages.
+## Why is it better to use Heroku for deploying your app?
+* #### **Service type: Pure PaaS**.
```
Comment (eliascodes) [inline coursebook/week-4/Heroku Workshop.md]: Is this supposed to mean that heroku doesn't provide non-PaaS related services? Is the point you were making that it's a relatively simple service compared to other PaaS vendors? If so, it's perhaps better to say that.

File: coursebook/week-4/Heroku Workshop.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+* #### **Ease of use:**  You can be up and running with an app continuously deployed from Github, for free, within minutes. Beautiful interface pared down to the bare essentials.
+
+* #### **Performance and Value: Pretty good compared to the other PaaS solutions, but still far off from IaaS.**
+
+* #### **Scalabilility: It really does not get easier than this. Adjust a slider (or have it scaled automatically) and handle all of the load balancing for you.**
```
Comment (eliascodes) [inline coursebook/week-4/Heroku Workshop.md]: Are beginners (or those with English as a second language) likely to understand what load balancing is?

File: coursebook/week-4/Heroku Workshop.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+```
+$ git push heroku master
+```
+
+
+The application now should be deployed.
+****
+
+### 4. Declare app dependencies
+Heroku recognizes an app as Node.js by the existence of a package.js file in the root directory. For your own apps, you can create one by running `npm init --yes`.
```
Comment (eliascodes) [inline coursebook/week-4/Heroku Workshop.md]: "package.js" -> "package.json"

File: coursebook/week-4/Heroku Workshop.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
+# Deploying Node.js Apps on Heroku
+***This Workshop assumes that you have a free Heroku account, and that you have Node.js and npm installed locally.***
+****
+## What is Heroku?
+Heroku is a cloud platform as a service (PaaS) supporting several programming languages that is used as a web application deployment model. Heroku supports Ruby, Java, Node.js, Scala, Clojure, Python and PHP programming languages.
```
Comment (eliascodes) [inline coursebook/week-4/Heroku Workshop.md]: "...cloud platform as a service (PaaS) **provider** supporting..."

File: coursebook/week-4/Heroku Workshop.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+## Why is it better to use Heroku for deploying your app?
+* #### **Service type: Pure PaaS**.
+
+* #### **Ease of use:**  You can be up and running with an app continuously deployed from Github, for free, within minutes. Beautiful interface pared down to the bare essentials.
```
Comment (eliascodes) [inline coursebook/week-4/Heroku Workshop.md]: "continuously deployed": will this be understood by beginners? We could possibly just drop the "continuously".
"pared down": will this be understood by those with English as a second language?

File: coursebook/week-4/Heroku Workshop.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+```
+Now check that you have git installed. If not, install it and test again.
+```
+$ git --version
+>
+git version 2.2.1
+```
+****
+### 2. Prepare the app
+You should have been cloned the [Node-girls-Workshop](https://github.com/node-girls/node-workshop) and followed the steps to make your own Node.js app.
```
Comment (eliascodes) [inline coursebook/week-4/Heroku Workshop.md]: "You should have been cloned" -> "You should have cloned"

File: coursebook/week-4/Heroku Workshop.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+```
+cd workshop-cms
```
Comment (eliascodes) [inline coursebook/week-4/Heroku Workshop.md]: This command assumes the name of the directory the app has been cloned into. Can we just instruct them to `cd` into the directory their node app is in?

**Comment by ronanyeah** [comment]
This workshop should be made into a repo: https://github.com/foundersandcoders/master-reference/blob/master/curriculum-planning/creation-of-resources.md

**Comment by eliascodes** [comment]
@sajedazoabi Please given an indication of whether you wish to keep this PR open to make the requested changes. If there's no response in within a week I'll assume there won't be any further work and close the PR. 

You are of course free to re-open the PR if you later change your mind 😃

**Pull request closed without merging**